Pod/Classes/UISideMenuNavigationController.swift

```diff
@@ -147,17 +147,52 @@ open class UISideMenuNavigationController: UINavigationController {
         if SideMenuManager.menuAllowPopIfPossible {
             for subViewController in navigationController.viewControllers {
                 if type(of: subViewController) == type(of: viewController) {
+                    if SideMenuManager.menuTabMode {
+                        navigationController.popToViewController(subViewController, animated: false)
+                        CATransaction.commit()
+                        return
+                    }
                     navigationController.popToViewController(subViewController, animated: animated)
                     CATransaction.commit()
                     return
                 }
             }
         }
         
+        if SideMenuManager.menuPreserveViewOnPush {
+            var viewControllers = navigationController.viewControllers
+            let filtered = viewControllers.filter{$0.restorationIdentifier == viewController.restorationIdentifier}
+            if let preservedView = filtered.first {
+                viewControllers = viewControllers.filter() { $0 !== preservedView }
+                viewControllers.append(preservedView)
+                if SideMenuManager.menuTabMode {
+                    navigationController.setViewControllers(viewControllers, animated: false)
```

Comment:
My understanding of this property is that when `menuReplaceOnPush = true`, you want to keep view controllers in memory but not in the visual stack (where the user can hit the back button). If this is true, then modifying the `viewControllers` property is probably not what you want, since the default behavior is to show back arrows, right? Unless you're expecting the developer to create a `leftBarButtonItem` on each screen to override it?